R/Design-methods.R

```diff
@@ -1,165 +1,151 @@
-# nolint start
-#####################################################################################
-## Author: Daniel Sabanes Bove [sabanesd *a*t* roche *.* com]
-##         Wai Yin Yeung [w*.* yeung1 *a*t* lancaster *.* ac *.* uk]
-## Project: Object-oriented implementation of CRM designs
-##
-## Time-stamp: <[Design-methods.R] by DSB Son 03/05/2015 20:35>
-##
-## Description:
-## Simulate outcomes from a CRM trial, assuming a true dose-toxicity
-## relationship.
-##
-## History:
-## 12/02/2014   file creation
-## 07/04/2014   start with parallelization on cores
-## 02/01/2015   rename: simulate.R --> Design-methods.R
-## 10/07/2015   added simulate methods
-#####################################################################################
-
-##' @include Data-methods.R
-##' @include Design-class.R
-##' @include McmcOptions-class.R
-##' @include Rules-methods.R
-##' @include Simulations-class.R
-##' @include helpers.R
-##' @include mcmc.R
-{}
-
-##' Helper function to set and save the RNG seed
-##'
-##' This is basically copied from simulate.lm
-##'
-##' @param seed an object specifying if and how the random number generator
-##' should be initialized (\dQuote{seeded}). Either \code{NULL} (default) or an
-##' integer that will be used in a call to \code{\link{set.seed}} before
-##' simulating the response vectors. If set, the value is saved as the
-##' \code{seed} slot of the returned object. The default, \code{NULL} will
-##' not change the random generator state.
-##' @return The RNGstate will be returned, in order to call this function
-##' with this input to reproduce the obtained simulation results
-##'
-##' @export
-##' @keywords programming
-##' @author Daniel Sabanes Bove \email{sabanesd@@roche.com}
-setSeed <- function(seed = NULL) {
+#' @include Data-methods.R
+#' @include Design-class.R
+#' @include McmcOptions-class.R
+#' @include Rules-methods.R
+#' @include Simulations-class.R
+#' @include helpers.R
+#' @include mcmc.R
+NULL
+
+# helper functions ----
+
+## set_seed ----
+
+#' Helper function to set and save the RNG seed
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This code is basically copied from `stats:::simulate.lm`.
+#'
+#' @param seed an object specifying if and how the random number generator
```

Comment:
```suggestion
#' @param seed (`integer` or `NULL`)\cr specifies if and how the random number generator
```

Reply:
One question: Do we want to have seed as integer? set.seed() accepts any number and interpret it as integer, so we should probably accept the same?

Reply:
I would say here `integer` (that is also in the sentence afterwards) and we can silently convert numeric to integer inside (with `safeInteger()`)

Reply:
OK. Just tested `safeInteger()` to convert a numeric value to an integer. The function did not convert e.g. `1.1` to `1`.  I am not sure how the function works as I expect.

Reply:
I think safeInteger() only converts values that are `is.wholenumber(x)`. I will convert seed directly with `as.integer` here.